File: 2015/test/day_10_test.dart
import 'package:test/test.dart';
import 'package:advent_2015/day_10.dart';

void main() {
  String testInput = "111221";
  String realInput = "3113322113";

  test('Part 1', () {
    expect(partOne(testInput), "312211");

    String answer = realInput;

    for (var x = 0; x < 40; x++) {
      answer = partOne(answer);
    }
    print(answer.length);
  });

  test('Part 2', () {
    expect(partOne(testInput), "312211");

    String answer = realInput;

    for (var x = 0; x < 50; x++) {
      answer = partOne(answer);
    }
    print(answer.length);
  });
}

File: 2024/lib/day_10_test.rb
require "minitest/autorun"

require_relative "./day_10"
require_relative "./helpers"

class Day10Test < Minitest::Test
  def self.test_order; :alpha; end # Stable test order for readability

  def setup
    @test_input = <<~INPUT
      89010123
      78121874
      87430965
      96549874
      45678903
      32019012
      01329801
      10456732
    INPUT

    @test_input = @test_input.strip

    @input = AdventOfCodeHelpers.get_input(10)
  end

  def test_part_1
    assert_equal 36, Day10.part_1(@test_input)

    answer = Day10.part_1(@input)
    puts "Part 1: #{answer}"

    AdventOfCodeHelpers.submit_answer(10, answer, 1)
  end

  def test_part_2
    assert_equal 81, Day10.part_2(@test_input)

    answer = Day10.part_2(@input)
    puts "Part 2: #{answer}"

    AdventOfCodeHelpers.submit_answer(10, answer, 2)
  end
end
